contracts/contracts/strategies/NativeStaking/ValidatorRegistrator.sol

```diff
@@ -0,0 +1,350 @@
+    uint256 public stakeETHTally;
+    // For future use
+    uint256[47] private __gap;
+
+    enum VALIDATOR_STATE {
+        NON_REGISTERED, // validator is not registered on the SSV network
+        REGISTERED, // validator is registered on the SSV network
+        STAKED, // validator has funds staked
+        EXITING, // exit message has been posted and validator is in the process of exiting
+        EXIT_COMPLETE // validator has funds withdrawn to the EigenPod and is removed from the SSV
+    }
+
+    event RegistratorChanged(address indexed newAddress);
+    event StakingMonitorChanged(address indexed newAddress);
+    event ETHStaked(
+        bytes32 indexed pubKeyHash,
+        bytes pubKey,
+        uint256 amount,
+        bytes withdrawal_credentials
+    );
+    event SSVValidatorRegistered(
+        bytes32 indexed pubKeyHash,
+        bytes pubKey,
+        uint64[] operatorIds
+    );
+    event SSVValidatorExitInitiated(
+        bytes32 indexed pubKeyHash,
+        bytes pubKey,
+        uint64[] operatorIds
+    );
+    event SSVValidatorExitCompleted(
+        bytes32 indexed pubKeyHash,
+        bytes pubKey,
+        uint64[] operatorIds
+    );
+    event StakeETHThresholdChanged(uint256 amount);
+    event StakeETHTallyReset();
+
+    /// @dev Throws if called by any account other than the Registrator
+    modifier onlyRegistrator() {
+        require(
+            msg.sender == validatorRegistrator,
+            "Caller is not the Registrator"
+        );
+        _;
+    }
+
+    /// @dev Throws if called by any account other than the Staking monitor
+    modifier onlyStakingMonitor() {
+        require(msg.sender == stakingMonitor, "Caller is not the Monitor");
+        _;
+    }
+
+    /// @dev Throws if called by any account other than the Strategist
+    modifier onlyStrategist() {
+        require(
+            msg.sender == IVault(VAULT_ADDRESS).strategistAddr(),
+            "Caller is not the Strategist"
+        );
+        _;
+    }
+
+    /// @param _wethAddress Address of the Erc20 WETH Token contract
+    /// @param _vaultAddress Address of the Vault
+    /// @param _beaconChainDepositContract Address of the beacon chain deposit contract
+    /// @param _ssvNetwork Address of the SSV Network contract
+    /// @param _maxValidators Maximum number of validators that can be registered in the strategy
+    constructor(
+        address _wethAddress,
+        address _vaultAddress,
+        address _beaconChainDepositContract,
+        address _ssvNetwork,
+        uint256 _maxValidators
+    ) {
+        WETH_TOKEN_ADDRESS = _wethAddress;
+        BEACON_CHAIN_DEPOSIT_CONTRACT = _beaconChainDepositContract;
+        SSV_NETWORK_ADDRESS = _ssvNetwork;
+        VAULT_ADDRESS = _vaultAddress;
+        MAX_VALIDATORS = _maxValidators;
+    }
+
+    /// @notice Set the address of the registrator which can register, exit and remove validators
+    function setRegistrator(address _address) external onlyGovernor {
```

Comment:
In the next several functions, the emits should be after the variable sets.

Reply:
fixed